### DISCUSSION TITLE
API endpoint to create sysadmin

### DISCUSSION THREAD

**Discussion by dhesru** [ckan/ckan#6068] [Ideas]
Currently creating/updating of sysadmin account is performed via a command line. Can this functionality be extended to allow user to create sysadmin accounts using API endpoints? 
This will give the users more flexibility when setting up a new application or even to maintain an existing one.

**Comment by wardi** [comment]
You can set `"sysadmin": true` on an existing user account with `user_update` or on a new account with `user_create`

**Comment by amercader** [comment]
For completeness, the next version of CKAN (current master) contains a new UI feature to add sysadmins from the frontend. This is not available on 2.9 but could be easily extracted to an extension.

https://github.com/ckan/ckan/pull/5522

![Screenshot_2021-05-04 Administration - CKAN](https://user-images.githubusercontent.com/200230/116987512-6fde9d00-accf-11eb-942a-688eb923da3d.png)

**Reply by jqnatividad** [reply]
I always thought sysadmin privs being available only from the CLI as a feature not a bug :), as IMHO, sysadmin rights should be given sparingly.

Not even functional sysadmins for CKAN sites that we helped manage are given sysadmin privs by default, as we had several painful "foot-shooting" episodes that required recovering from backup.

To that point, it would be nice if there is:
* configuration option to switch off the sysadmin privs UI
* authentication logging (logfile and/or Activity Stream) of not only privilege activity (including Editor and Membership rights grants/revokes), but user activity as well (logins and authentication failures).
